### PR TITLE
Documentation for custom rules and closure rules

### PR DESCRIPTION
**Pull request by MissaelAnda**
## Document Custom Rule Classes and Closure Rules

Add a description to Closure rule with a DocComent:
```php
$request->validate([
    'complex_rule' => [
        /** This is a very specific validation therefore it needs a better description */
        function ($attribute, $value, $fail) { $fail('error'); }
    ],
]);
```

Add more info to a Custom Validation Rule:
```php
class CustomRule implements Rule
{
    // ...

    public static function docs(): array
    {
        return [
            'description' => 'This rule makes something very specific',
            'example' => '52KB',
        ];
    }
}
```

This is a followup for #551

I'm missing the tests and documentation changes, if you could point me where and how should I do them it'd be nice.

### CODE REVIEW

**Comment by shalvah** [comment]
The class that tests this is the `ValidationRuleParsingTest`. You should probably add two new test methods, something like `can_parse_custom_closure_rules` and `can_parse_custom_rule_classes`. You can follow the existing examples.

**Comment by shalvah** [comment]
It feels a bit iffy to me to have an example defined on a rule. I guess it can work, but I'm uncertain if we want to go that path. For example, what happens if one parameter has this rule and others defined? The example will either be overwritten or hardcoded to this, which may or may not be unexpected behaviour. Maybe we should leave that out for now (after all, you can already specify examples for parameters in a FormRequest).

**Comment by MissaelAnda** [comment]
The idea with the example is to define a default example in case none given like the string rules "setter"s, the problem maybe is that the example is always assigned. I'll change it to only set the example with the "setter" field and add the tests.

File: src/Extracting/ParsesValidationRules.php [hunk 1]
```diff
@@ -164,300 +166,328 @@ protected function normaliseRules(array $rules): array
...
-            }
+                $dependentRules = ['between', 'max', 'min', 'size', 'gt', 'gte', 'lt', 'lte', 'before', 'after', 'before_or_equal', 'after_or_equal'];
+                if ($independentOnly && in_array($rule, $dependentRules)) {
+                    return false;
+                }
 
-            return true;
-        } catch (Throwable $e) {
-            throw CouldntProcessValidationRule::forParam($parameterData['name'], $rule, $e);
+                // Reminders:
```
Comment (shalvah) [inline src/Extracting/ParsesValidationRules.php:210]: Btw, we can use early returns here to avoid this huge diff from indentation.

Comment (shalvah) [reply src/Extracting/ParsesValidationRules.php]: Done.

**Comment by shalvah** [comment]
Can you send in a docs PR as well?

**Comment by MissaelAnda** [comment]
I just relized that I'm missing `InvokableRule`, it only needs to change `ParsesValidationRules` to import it and change line 194 to:
```php
if ($rule instanceof Rule || $rule instanceof InvokableRule) {
```

**Comment by shalvah** [comment]
Added.

**Pull request merged**